lib/src/main/java/com/androidessence/materialdesignspecs/BaseColorAdapter.java

```diff
@@ -0,0 +1,118 @@
+package com.androidessence.materialdesignspecs;
+
+import android.content.Context;
+import android.support.annotation.LayoutRes;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import com.androidessence.materialdesignspeclibrary.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by ankitagrawal on 5/15/17.
+ */
+@SuppressWarnings("WeakerAccess")
+public abstract class BaseColorAdapter extends RecyclerView.Adapter<BaseColorAdapter.BaseColorViewHolder> {
+    private List<Integer> colors = new ArrayList<>();
+    private int selectedPosition = -1;
```

Comment:
well the only reason its only at one place because sometimes getAdapterPosition() returns RecyclerView.NO_POSITION , although the value is -1 currently it may not be so in future, but still anyone i have changed that magic number